ghc-hotswap/GHC/Hotswap.hs

```diff
@@ -122,22 +122,21 @@ updateState
   -> FilePath          -- path to the next instance of the shared object
   -> IO ()
 updateState mvar symbolName nextPath = do
-  newVal <- force <$> loadNewSO symbolName nextPath
-  -- Build a new state for this version
-  newLock <- L.new
-  let
-    newState = SOState
-      { lock = newLock
-      , path = nextPath
-      , val = newVal
-      }
-  -- Swapping in the new state means all new calls to `withSO` from the client
-  -- will use the new value. After this it's impossible for a new read lock to
-  -- grab the old state
-  oldState <- swapMVar mvar newState
-  -- All readers in oldState will fall out, so we're safe to destroy state here
-  L.withWrite (lock oldState) $
-    unloadObj (path oldState)
+  modifyMVar_ mvar
```

Comment:
Ah, I see what you did, but this has other consequences. Pushing all of the IO actions inside of `modifyMVar_` here will cause all your incoming requests to block until this update finishes. As an update can take multiple seconds, that means anything using the data underneath this would stall for multiple seconds. In the use-case we have, this would be a non-starter.